### ISSUE TITLE
UCM fails if driver built into kernel

### ISSUE THREAD

**Issue by caseorum** [alsa-project/alsa-ucm-conf#52]
Since `class/sound/card${CardNumber}/device/driver/module` does not exist if the module is built-in, `${sys:$KernelModulePath}` is left undefined.
This is caused by b3531451db3b4ac6aa8c6a8bec3992bcf30fffe4. The fallback suggested in alsa-project/alsa-lib#79 works for at least my bdw-rt5677.
Is there a more general solution here? Maybe `.../device/modalias`?

**Comment by plbossart** [comment]
@caseorum we typically don't have any audio drivers compiled as built-in due to dependencies on firmware and topology files. Wondering what configuration you used to reach this problem?

**Comment by caseorum** [comment]
@plbossart Everything built-in, zero kernel dependencies on userspace. The same I described in https://github.com/thesofproject/sof/issues/2459#issuecomment-696301613. I'm now solving the "UCM weirdness" I alluded to. It looks like NXP cares about this kind of boot too (https://github.com/thesofproject/linux/issues/139), but I don't know if they use UCM.

**Comment by perexg** [comment]
The bdw-rt5677 configuration is not converted to the modules layout, so the fallback to the LongName / CardDriver should be active. What does say 'alsaucm' command? Do you see an error?

**Comment by caseorum** [comment]
@perexg The suggested fallback was never merged. Reverting b3531451db3b4ac6aa8c6a8bec3992bcf30fffe4 fixes for bdw-rt5677 but it's not a general solution. Current master:
```
[user@localhost alsa-ucm-conf-git]$ alsaucm -c SOF list _verbs
ALSA lib ucm_subs.c:367:(uc_mgr_get_substituted_value) variable '${sys:$KernelModulePath}' is not defined in this context!
ALSA lib main.c:983:(snd_use_case_mgr_open) error: failed to import SOF use case configuration -22
alsaucm: error failed to open sound card SOF: Invalid argument
```

**Comment by perexg** [comment]
Could you try this small configuration change ?

```
diff --git a/ucm2/ucm.conf b/ucm2/ucm.conf
index ed410dd..9e78df1 100644
--- a/ucm2/ucm.conf
+++ b/ucm2/ucm.conf
@@ -52,7 +52,7 @@ If.driver {
                        }
                        False {
                                Define.KernelModulePath "class/sound/card${CardNumber}/device/driver/module"
-                               Define.KernelModule "${sys:$KernelModulePath}"
+                               Define.KernelModule "$${sys:$KernelModulePath}"
                                UseCasePath.module {
                                        Directory "module"
                                        File "${var:KernelModule}.conf"
```

**Comment by caseorum** [comment]
@perexg That's it! Awesome.

**Comment by perexg** [comment]
Thanks for the bug report. Fixed in 5634055371fe6b46460502a136f09f20a7318940 .
